src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Utility.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.EventHubs;
+using Microsoft.Azure.EventHubs.Processor;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.EventHubs
+{
+    internal class Utility
+    {
+        public static void LogException(Exception ex, string message, ILogger logger)
+        {
+            try
+            {
+                // Sometimes EventHub SDK agrigates an exception
+                AggregateException ae = ex as AggregateException;
+                if (ae != null && ae.InnerExceptions != null && ae.InnerExceptions.Count == 1)
+                {
+                    ex = ae.InnerExceptions[0];
+                }
+
+                LogLevel logLevel = GetLevel(ex);
+                if (logLevel == LogLevel.Information)
+                {
+                    message = $"{message} An exception of type '{ex.GetType().Name}' was thrown. This exception type is typically a result of Event Hub processor rebalancing or a transient error and can be safely ignored.";
+                }
+                logger?.Log(logLevel, 0, message, ex, (s, exc) => message);
+            }
+            catch
+            {
+                // best effort logging
+            }
+        }
+
+        private static LogLevel GetLevel(Exception ex)
+        {
+            if (ex is ReceiverDisconnectedException || ex is LeaseLostException
+                || (ex is WindowsAzure.Storage.StorageException && ex.Message == "The lease ID specified did not match the lease ID for the blob."))
```

Comment:
Rather than check the message -- you can do this, which should be safer: https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Shared.Storage/StorageExceptionsExtensions.cs#L173.